internal/controller/instaslice_controller.go

```diff
@@ -523,9 +526,27 @@ func (r *InstasliceReconciler) setupWithManager(mgr ctrl.Manager) error {
 	if err != nil {
 		return err
 	}
+	// pod filtering
+	// only watches Pods that:
+	// have the mutation annotation, or
+	// have meaningful Change
+	instaslicePredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return hasInstasliceMutation(e.Object)
+		},
+		// Pod was already mutated, and instaslice is managing it
+		// Something meaningful changed, and need to reconcile
+		UpdateFunc: func(e event.UpdateEvent) bool {
```

Comment:
I think we could use a LabelSelectorPredicate here.

Reply:
We are using an annotation, not a label. LabelSelectorPredicate won't match anything unless we switch the webhook to use a label.

Reply:
I will switch to a label so we can use LabelSelectorPredicate